### ISSUE TITLE
ConfigValueFactory should accept collections containing ConfigValues

### ISSUE THREAD

**Issue by argv-minus-one** [typesafehub/config#179]
For some reason, the methods of the `ConfigValueFactory` do not seem to handle values that are already `ConfigValue`s. For instance:

```scala
import com.typesafe.config._
import scala.collection.JavaConverters._

val a = ConfigValueFactory.fromAnyRef(42)
ConfigValueFactory.fromMap(Map("a" -> a).asJava)
```

…yields the exception…

```
com.typesafe.config.ConfigException$BugOrBroken: bug in method caller: not valid to create ConfigValue from: ConfigInt(42)
  at com.typesafe.config.impl.ConfigImpl.fromAnyRef(ConfigImpl.java:275)
  at com.typesafe.config.impl.ConfigImpl.fromAnyRef(ConfigImpl.java:253)
  at com.typesafe.config.impl.ConfigImpl.fromAnyRef(ConfigImpl.java:193)
  at com.typesafe.config.ConfigValueFactory.fromAnyRef(ConfigValueFactory.java:66)
  at com.typesafe.config.ConfigValueFactory.fromMap(ConfigValueFactory.java:94)
  at com.typesafe.config.ConfigValueFactory.fromMap(ConfigValueFactory.java:134)
  ... 63 elided
```

Shouldn't that be valid?

**Comment by havocp** [comment]
This was fixed in b81a45870d9395c62e1f4f8228ab2b49f1d1f167 I believe.

**Comment by alivcor** [comment]
For anyone who lands up on this page, if you're using Scala, make sure to do `.asJava` on ALL the `Map` objects you create.
